From 759d191c589f9390b2d1f5416604568b2f2cfdb7 Mon Sep 17 00:00:00 2001
From: Denis Barbier <bouzim@gmail.com>
Date: Wed, 29 Oct 2014 23:10:39 +0000
Subject: [PATCH] Fix build failures with Mesa 10.3.1
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Newer Mesa looks buggy, several errors like this one are reported:
 /usr/include/GL/glxext.h:480:143: error: ‘GLintptr’ has not been declared
 typedef void ( *PFNGLXCOPYBUFFERSUBDATANVPROC) (Display *dpy, GLXContext readCtx, GLXContext writeCtx, GLenum readTarget, GLenum writeTarget, GLintptr readOffset, GLintptr writeOffset, GLsizeiptr size);
Work around this issue by defining GLX_GLXEXT_LEGACY.

Fix issue #534.
---
 src/OpenGl/OpenGl_GlCore11.hxx | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/OpenGl/OpenGl_GlCore11.hxx b/src/OpenGl/OpenGl_GlCore11.hxx
index fe76d31..e14c020 100644
--- a/src/OpenGl/OpenGl_GlCore11.hxx
+++ b/src/OpenGl/OpenGl_GlCore11.hxx
@@ -36,6 +36,7 @@
 
 // exclude modern definitions and system-provided glext.h, should be defined before gl.h inclusion
 #define GL_GLEXT_LEGACY
+#define GLX_GLXEXT_LEGACY
 
 // include main OpenGL header provided with system
 #if defined(__APPLE__) && !defined(MACOSX_USE_GLX)
